packages/build/src/signable-compiler.ts

```diff
@@ -57,13 +35,8 @@ class SignableCompiler extends Compiler {
    * @returns {string} The target name.
    */
   determineTarget(): string {
-    switch(this.platform) {
-      case Platform.Windows: return Target.Windows;
-      case Platform.MacOs: return Target.MacOs;
-      default: return Target.Linux;
-    }
+    return '^12.0.0';
```

Comment:
> Do we want to make this part of the build config?

I think it would be very nice.

> "determineTarget()" and "Is the lack of cross-“compilation” an issue here"

Kind of, but not only. I think that can be changed anyway. 

We had some initial struggle to tame evergreen. That brought us to rely a lot on what we knew we could rely on, and to have this "detection of platform" approach in all sort of places. In all fairness that has the great merit of having brought us this far and have a release available in all the channels we have to.

On a less bright note, that creates a fair amount of non-obvious coupling, It would be really nice in a future iteration to make use of what we now know evergreen can offer, and have a more explicit way to instruct the CI what to build and when.


@lrlna i think you have most context here.

Reply:
> > Do we want to pin a specific version
> 
> I'd really recommend to have the same version both here and in `.evergreen/.install_node` and `InstallNode.ps1`

Okay, that’s a good point :+1: But still, do we want a specific version or just “latest v12.x/latest v14.x”?

Reply:
> @lrlna i think you may have more context here.

`determineTarget()` in this particular case was only used for `pkg` and `nexe`. So, if a binary that `boxednode` builds can be used to package for `.tgz`, `.deb` and `.rpm`, we do not need this information.
> "detection of platform" approach in all sort of places
We use `os.platform()` and build variant information (`rhel`, `macos`, `debian` etc) when creating a tarball/archive. I haven't found a better way for us to use this from the CI directly, which is why we are detecting current platform or build variant in `tarball.ts`